viewer/src/parser.ts

```diff
@@ -19,20 +21,36 @@ export class Parser {
     private finalStates: Array<String>;
     private entryPointState: string;
     private entryPointStateMachine: string;
-    private stateMachineNames: Array<string>;
     private linkDataArray: Array<LinkDataArrayTemplate>;
     private nodeDataArray: Array<NodeDataArrayTemplate>;
 
+    public promisifyParserString: any;
+
     constructor(componentGraphicalModel: ComponentGraphicalModel) {
         this.componentGraphicalModel = componentGraphicalModel;
+        this.promisifyParserString = promisify(parseString);
+    }
+
+    parse(parseListener: (error: Error, parser: Parser) => void) {
```

Comment:
Why do you need to have parser as parameter of your callback? Why not just the result of the parsing ? no?
Perhaps returning a promise could be a better choice than using callbacks